File: CRM/Models/ViewModels/LeadAssignmentSelectedPartnerViewModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace CRM.Models.ViewModels
{
    public class LeadAssignmentSelectedPartnerViewModel
    {
        public Guid LeadId { get; set; }

        public List<Guid> PartnerBranchIds { get; set; }
    }
}

File: CRM/wwwroot/js/viewmodels/LeadAssignmentSelectedPartnerViewModel.js
class LeadAssignmentSelectedPartnerViewModel {
    constructor(leadId) {
        this.LeadId = leadId;
        this.PartnerBranchIds = [];
    }

    addPartner(branchId) {
        this.PartnerBranchIds.push(branchId);
    }
}